src/main/java/bio/terra/workspace/service/resource/controlled/flight/CreateControlledResourceFlight.java

```diff
@@ -31,10 +33,16 @@ public CreateControlledResourceFlight(FlightMap inputParameters, Object applicat
     // create the cloud resource via CRL
     final ControlledResource resource =
         inputParameters.get(JobMapKeys.REQUEST.getKeyName(), ControlledResource.class);
+    final AuthenticatedUserRequest userRequest =
+        inputParameters.get(JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
 
     switch (resource.getResourceType()) {
       case GCS_BUCKET:
-        addStep(new CreateGcsBucketStep(flightBeanBag.getCrlService()));
+        addStep(
+            new CreateGcsBucketStep(
+                flightBeanBag.getCrlService(),
+                (ControlledGcsBucketResource) resource,
```

Comment:
(nit) any reason not to do this cast above where the variable is defined? And how are you confident it will succeed?

Reply:
Above we don't know which resource type we have. We check that in the switch, so I know this will succeed on this branch.